compiler/rustc_codegen_ssa/src/back/linker.rs

```diff
@@ -1831,12 +1832,13 @@ impl<'a> Linker for PtxLinker<'a> {
     }
 
     fn optimize(&mut self) {
-        match self.sess.lto() {
-            Lto::Thin | Lto::Fat | Lto::ThinLocal => {
-                self.cmd.arg("-Olto");
-            }
-
-            Lto::No => {}
+        match self.sess.opts.optimize {
+            OptLevel::No => "-O0",
+            OptLevel::Less => "-O1",
+            OptLevel::Default => "-O2",
+            OptLevel::Aggressive => "-O3",
+            OptLevel::Size => "-Os",
+            OptLevel::SizeMin => "-Oz",
```

Comment:
I'm very much in favor of adding the new linker flavor from the start.

With a new linker flavor we don't need to worry about breakage between rust-ptx-linker and the new `llvm-bitcode-linker`. When additional target reach some sort of support they can be used with the new linker using `-C linker-flavor llbc` instead of the misleading `-C linker-flavor ptx`. It should be possible to deprecate the ptx-linker flavor quite quickly, but I don't know what's actually required to get rid of it for good.

I see that there is a concept of unstable linker flavor. Since the goal is to distribute `llvm-bitcode-linker` versioned together with rust it would be beneficial to mark it as unstable and avoid making it default for any target yet. This means only (as default) using it for ptx tests at this point, but start recommending it for experimentation for the nvptx target as soon as it can be distributed to users.